src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+  const [selectedItem, setSelectedItem] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(result => {
+        setTodos(result);
+        setLoading(false);
+      })
+      .catch(() => {
+        throw new Error('Uploading error');
+      });
```

Comment:
`throw new Error('Uploading error');` make sense to use if you have some conditions at `.then` statement and wanna  throw an Error that you catch at `.catch` statement

Also you need to stop your loader not depends on result
```suggestion
      .then(result => {
        setTodos(result);
      })
      .catch((error) => {
        console.error(error)
      })
      .finally(() => {
        setLoading(false);
      });
```